src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -1012,7 +1012,14 @@ primitive_inst::primitive_inst(network& network, program_node const& node, bool
                 if (user->is_type<mutable_data>())
                     _outputs[0] = user->as<mutable_data>().get_attached_memory_ptr();
         } else {
-            _outputs = allocate_outputs();
+            auto& e = network.get_engine();
+            auto output_layout = node.get_output_layout();
+            // When dynamic shape node has huge upper boundary which causes bigger mem size than system max allocable mem size, return false.
+            if (!e.check_allocatable(output_layout, e.get_lockable_preferred_memory_allocation_type(output_layout.format.is_image_2d()))) {
```

Comment:
1. I think you can just pass usm_host to this check. (actual alloc type will be checked again in allocate_output but here, we just filter out the tensors which cannot be allocated even for host) 
2. Also we need to do this check only for dynamic shape, because static shape check will be done in allocate_output. 
3. So my suggestion is
-  // not to set _mem_allocated in the constructors initializer (get rid of line 978) 
- if (node.is_dynamic() && (!check_allocatable(output_layout, usm_host)) 
        allocate_memory = false;
   _mem_allocated = allocate_memory;
   if (allocate_memory) {
        // original code
   }

Reply:
Updated! Thanks!